pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2567,6 +2567,26 @@ type StorageScopeSpec struct {
 	// +nullable
 	// +optional
 	StorageClassDeviceSets []StorageClassDeviceSet `json:"storageClassDeviceSets,omitempty"`
+	// +optional
+	Store OSDStore `json:"store,omitempty"`
+}
+
+// StoreType is the backend storage type for OSDs
+type StoreType string
```

Comment:
Is the StoreType really useful? Since we have the enum in the CRD validation, the user won't be able to specify invalid values anyway. Just using a string directly seems simpler so we don't have to cast it back to a string.

Reply:
It can be useful within the code base to refer to the store types, like cephv1.Bluestore or cephv1.Bluestore-rdr.  What do you think?

Reply:
String constants could still be defined, without defining a new `StoreType`.

Reply:
Moved it as string constants within the `storage` API so that it can be accessed in other modules easily.